src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2516,6 +2755,12 @@ ostree_sysroot_write_deployments_with_options (OstreeSysroot     *self,
   if (!_ostree_sysroot_ensure_writable (self, error))
     return FALSE;
 
+  /* for now, this is gated on an environment variable */
+  const char *opted_in = getenv ("OSTREE_ENABLE_AUTO_EARLY_PRUNE");
```

Comment:
We do have `OSTREE_SYSROOT_OPTS` today which is basically intended for things like this.

So it'd look more like `OSTREE_SYSROOT_OPTS=early-prune`

Reply:
Yeah, much nicer!